pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -754,15 +755,52 @@ func (s *FlinkStateMachine) handleSubmittingJob(ctx context.Context, app *v1beta
 	if job == nil {
 		return statusUnchanged, errors.Errorf("Could not find job %s", s.flinkController.GetLatestJobID(ctx, app))
 	}
-
-	// wait until all vertices have been scheduled and started
-	allVerticesStarted := true
-	for _, v := range job.Vertices {
-		allVerticesStarted = allVerticesStarted && (v.StartTime > 0)
+	allVerticesRunning := true
+	if app.Spec.BetaFeaturesEnabled {
```

Comment:
Are we documenting what features are in beta?

Reply:
This is not a standard pattern.

Primarily because, this flag getting updated will cause applications to restart.

Reply:
@sethsaperstein-lyft I will add more to crd.md file for documenting

Reply:
@anandswaminathan sorry, I didn't follow that why if the flag is updated, the application to restart. Can you elaborate that? My understanding is that this field is set in applicaiton jsonnet file and the value is passed in when new deployment/update is made. But this field can't be changed in runtime.

Reply:
@leoluoInSea 

So here is what you are doing - you are looking for a way to have some applications be exposed to this feature. 

This is not something that is commonly done in CRD's and operator, as this is not something you want to have for the long run. Say you remove the flag some day later, all the apps that have this set need to revert, and we are looking at updating 100s of apps (depending on scale). 

If you are worried about the reliability of the change, may be pass a version as label (if we already don't do so), and have logic on the label. That way, if the code gets removed, or changed, the apps will not need to be updated.

Reply:
Just for record, Anand and I had a discussion and our conclusion is that this is not breaking changes, and I will do several testing in staging after merging, definitely with an example of failure scenario `nmodes`. 

For future benefit if we need a feature flag, as Anand pointed out above, the recommended way is to check label instead of adding a new field in CRD. From app jsonnet file, just add the label if we need to turn on the feature.